### ISSUE TITLE
Horovod got different models

### ISSUE THREAD

**Issue by shunshunyin** [horovod/horovod#1121]
**Environment:**
1. Tensorflow
2. Tensorflow 1.12.0
3. cuda 9

**Your question:**
I run bert using horovod on three gpus, and got three models. Howerver, I got different predict result on the same test data, in other words, the three models are different, and the predict result diff is very small. Is this normal?

**Comment by alsrgv** [comment]
@shunshunyin, yes - it's expected to have a tiny divergence between models.  It could be caused by tiny numerical differences or different batchnorm stats. One thing you can do is to broadcast parameters every epoch (or other # of steps) - that would make the difference even smaller.
